javalanglib/src/main/scala/java/lang/Long.scala

```diff
@@ -114,6 +114,42 @@ object Long {
     }
   }
 
+  // Ported from https://github.com/gwtproject/gwt/blob/master/user/super/com/google/gwt/emul/java/lang/Long.java
+  def toString(value: scala.Long, intRadix: Int): String = {
+    if (intRadix == 10 || intRadix < Character.MIN_RADIX || intRadix > Character.MAX_RADIX) {
+      toString(value)
+    } else if (Integer.MIN_VALUE <= value && (value <= Integer.MAX_VALUE)) {
+      Integer.toString(value.toInt, intRadix)
+    } else {
+      // Character.forDigit checks the digit range which is not the requirement here - hence the apparent duplication
+      def forDigit(digit: Int): Char = {
+        val overBaseTen = digit - 10
+        val result = if (overBaseTen < 0) '0' + digit else 'a' + overBaseTen
+        result.toChar
+      }
+
+      // If the value is zero or positive, we can reduce code by negating the value for the string conversion
+      // but not adding a '-' at the end
+
+      var _value: scala.Long = if (value >= 0) -value else value
+      var res = new String
```

Comment:
np - the trouble with porting this code is that it's occasionally hard to remember what language I'm actually using - C, C++, Java or Scala!! Thanks for the reminder...